tv_apps/remote-control/js/config.js

```diff
@@ -0,0 +1,139 @@
+/* global SettingsListener, evt */
+'use strict';
+
+(function(exports) {
+  var PREFIX = 'remote-control.';
+  var SETTINGS = {
+    'enabled': false,
+    'pairing-required': false,
+    'authorized-devices': null,
+    'server-ip': ''
+  };
+
+  var DEBUG = false;
+
+  function Config() {
+    this._ready = false;
+    this._observer = [];
+    this.settings = [];
+  }
+
+  Config.prototype = evt({
+    start: function() {
+      for(var name in SETTINGS) {
+        this._observer[name] = this._settingHandler.bind(this, name);
+        SettingsListener.observe(PREFIX + name, SETTINGS[name],
+          this._observer[name]);
+      }
+    },
+
+    stop: function() {
+      for(var name in SETTINGS) {
+        SettingsListener.unobserve(PREFIX + name, this._observer[name]);
+      }
+      this.settings = [];
+      this._observer = [];
+      this._ready = false;
+    },
+
+    prepareOptions: function(callback) {
+      var checked;
+      if (this.settings.enabled) {
+        if (this.settings['pairing-required']) {
+          checked = 'option-pair-on';
+        } else {
+          checked = 'option-pair-off';
+        }
+      } else {
+        checked = 'option-disabled';
+      }
+
+      this.check(checked);
+
+      var authorizedDevices = [];
+      if (this.settings['authorized-devices']) {
+        authorizedDevices = JSON.parse(this.settings['authorized-devices']);
+      }
+      var canClear = Object.keys(authorizedDevices).length;
+      var button = document.getElementById('clear-paired-devices');
+      button.disabled = !canClear;
+      if (canClear) {
+        button.removeAttribute('aria-hidden');
+      } else {
+        button.setAttribute('aria-hidden', true);
+      }
+
+      callback(checked);
+    },
+
+    check: function(id) {
+      var radios = document.querySelectorAll('.option-radio');
+      Array.from(radios).forEach(function(radio) {
+        if (radio.id == id) {
+          radio.classList.add('checked');
+        } else {
+          radio.classList.remove('checked');
+        }
+      });
+    },
+
+    save: function(callback) {
+      var settings = {};
+      var checked = document.querySelector('.option-radio.checked').id;
+      switch(checked) {
+        case 'option-pair-on':
+          settings[PREFIX + 'enabled'] = true;
+          settings[PREFIX + 'pairing-required'] = true;
+          break;
+        case 'option-pair-off':
+          settings[PREFIX + 'enabled'] = true;
+          settings[PREFIX + 'pairing-required'] = false;
+          break;
+        case 'option-disabled':
+          settings[PREFIX + 'enabled'] = false;
```

Comment:
I feel we should give `pairing-required` a status although it doesn't hurt. I vote for `true`.